modules/enterprise-management/src/test/java/com/ibm/cloud/platform_services/enterprise_management/v1/EnterpriseManagementIT.java

```diff
@@ -1,433 +1,563 @@
+            .accountId(accountId)
+            .limit(limit)
+            .nextDocid(nextDocId)
+            .build();
+
+        // Invoke operation
+        Response<ListEnterprisesResponse> response = service
+            .listEnterprises(listEnterprisesOptions)
+            .execute();
+        // Validate response
         assertNotNull(response);
         assertEquals(response.getStatusCode(), 200);
-        AccountGroup responseObj = response.getResult();
-        assertNotNull(responseObj);
-        crn = responseObj.getCrn();
-    }
-
-    @Test
-    public void test15_UpdateAccountGroupWOptions() throws Throwable {
-        UpdateAccountGroupOptions updateAccountGroupOptions = new UpdateAccountGroupOptions.Builder()
-                .accountGroupId(accountGroupId).name("testString").build();
-
-        Response<Void> response = service.updateAccountGroup(updateAccountGroupOptions).execute();
-        assertNotNull(response);
-        assertEquals(response.getStatusCode(), 204);
-    }
 
-    @Test
-    public void test17_CreateVerificationCode() throws Throwable {
-        accountManagementService.createVerificationCode(email, token, config.get("AM_HOST"));
-    }
-
-    @Test
-    public void test18_GetVerificationCode() throws Throwable {
-        String response = accountManagementService.getVerificationCode(email, token, config.get("AM_HOST"));
-        JSONObject obj = new JSONObject(response);
-        JSONArray arr = obj.getJSONArray("resources");
-        verificationCode = arr.getJSONObject(0).getString("code");
-    }
-
-    @Test
-    public void test19_PutConfirmVerificationCode() throws Throwable {
-        accountManagementService.confirmVerificationCode(email, verificationCode, token, config.get("AM_HOST"));
-    }
-
-    @Test
-    public void test20_CreateStandardAccount() throws Throwable {
-        String response = accountManagementService.createStandardAccount(email, verificationCode, token,
-                config.get("AM_HOST"));
-        JSONObject obj = new JSONObject(response);
-        accountId = obj.getString("id");
-    }
-
-    @Test
-    public void test21_GetAccountByAccountIdLastOperation() throws Throwable {
-        accountManagementService.getAccountByAccountIdLastOperation(accountId, token, config.get("AM_HOST"));
-    }
-
-    @Test
-    public void test22_GetAccountActivationToken() throws Throwable {
-        URL url = null;
-        try {
-            url = new URL(config.get("DB_URL"));
-        } catch (MalformedURLException e) {
-            fail("The url is not well formed: " + url);
+        ListEnterprisesResponse listEnterprisesResponseResult = response.getResult();
+        assertNotNull(listEnterprisesResponseResult);
+
+        nextDocId = getNextDocId(listEnterprisesResponseResult.getNextUrl());
+        enterpriseList.addAll(listEnterprisesResponseResult.getResources());
+
+        log(String.format("Received a total of %d enterprises.%n", enterpriseList.size()));
```

Comment:
shouldn't this be after the end of the do/while loop?

Reply:
yikes...